django_mongodb_backend/schema.py

```diff
@@ -420,6 +423,79 @@ def _field_should_have_unique(self, field):
         # The _id column is automatically unique.
         return db_type and field.unique and field.column != "_id"
 
+    def _create_collection(self, model):
+        """
+        Create a collection for the model with the encrypted fields. If
+        provided, use the `_encrypted_fields_map` in the client's
+        `auto_encryption_opts`. Otherwise, create the encrypted fields map
+        with `_get_encrypted_fields_map`.
+        """
+        db = self.get_database()
+        db_table = model._meta.db_table
+        if model_has_encrypted_fields(model):
+            client = self.connection.connection
+            auto_encryption_opts = getattr(client._options, "auto_encryption_opts", None)
+            if not auto_encryption_opts:
+                raise ImproperlyConfigured(
+                    "Encrypted fields found but "
+                    f"DATABASES[{self.connection.alias}]['OPTIONS'] is missing "
+                    "auto_encryption_opts. Please set `auto_encryption_opts` "
+                    "in the connection settings."
+                )
+            encrypted_fields_map = getattr(auto_encryption_opts, "_encrypted_fields_map", None)
+            if not encrypted_fields_map:
+                encrypted_fields_map = self._get_encrypted_fields_map(
+                    model, client, auto_encryption_opts
+                )
+            else:
+                # If the encrypted fields map is provided, get the map for the
+                # specific collection.
+                encrypted_fields_map = encrypted_fields_map.get(db_table)
+            db.create_collection(db_table, encryptedFields=encrypted_fields_map)
+        else:
+            db.create_collection(db_table)
+
+    def _get_encrypted_fields_map(self, model, client, create_new_keys=False):
+        connection = self.connection
+        fields = model._meta.fields
+        options = client._options
+        auto_encryption_opts = options.auto_encryption_opts
+        kms_provider = router.kms_provider(model)
+        master_key = self.connection.settings_dict.get("KMS_CREDENTIALS").get(kms_provider)
+        client_encryption = ClientEncryption(
+            auto_encryption_opts._kms_providers,
+            auto_encryption_opts._key_vault_namespace,
+            client,
+            client.codec_options,
+        )
+        key_vault_db, key_vault_coll = auto_encryption_opts._key_vault_namespace.split(".", 1)
+        key_vault_collection = client[key_vault_db][key_vault_coll]
+        db_table = model._meta.db_table
+        field_list = []
+        for field in fields:
+            if getattr(field, "encrypted", False):
+                key_alt_name = f"{db_table}_{field.column}"
+                if not create_new_keys:
```

Comment:
You have `if not create_new_keys:` and `if not encrypted_fields_map:` in this file with else branches. I think it helps readability to remove the `not` and reverse the order of the branches.